src/include/miopen/graphapi/engineheur.hpp

```diff
@@ -84,16 +84,22 @@ class BackendEngineHeurDescriptor : public BackendDescriptor
 
     class OwnedEngineCfgDescriptor : public BackendEngineCfgDescriptor
     {
-    private:
+        using Base = BackendEngineCfgDescriptor;
         BackendEngineDescriptor mOwnedEngineDescriptorInstance;
 
     public:
-        OwnedEngineCfgDescriptor(EngineCfg&& engineCfg,
-                                 miopenBackendDescriptor_t opGraphDescriptor);
-        OwnedEngineCfgDescriptor(const OwnedEngineCfgDescriptor& other);
-        OwnedEngineCfgDescriptor(OwnedEngineCfgDescriptor&& other) noexcept;
-        OwnedEngineCfgDescriptor& operator=(const OwnedEngineCfgDescriptor& other);
-        OwnedEngineCfgDescriptor& operator=(OwnedEngineCfgDescriptor&& other) noexcept;
+        OwnedEngineCfgDescriptor(const EngineCfg& engineCfg,
+                                 miopenBackendDescriptor_t opGraphDescriptor)
+            : Base(engineCfg, &mOwnedEngineDescriptorInstance),
+              mOwnedEngineDescriptorInstance(Base::getEngineCfg().getEngine(), opGraphDescriptor)
+        {
+        }
+
+        OwnedEngineCfgDescriptor(const OwnedEngineCfgDescriptor& other)     = default;
+        OwnedEngineCfgDescriptor(OwnedEngineCfgDescriptor&& other) noexcept = default;
+        ;
+        OwnedEngineCfgDescriptor& operator=(const OwnedEngineCfgDescriptor& other) = default;
```

Comment:
```suggestion
        OwnedEngineCfgDescriptor(const OwnedEngineCfgDescriptor& other)     = delete;
        OwnedEngineCfgDescriptor(OwnedEngineCfgDescriptor&& other) noexcept = default;
        ;
        OwnedEngineCfgDescriptor& operator=(const OwnedEngineCfgDescriptor& other) = delete;
```
This introduces a bug: the value of `BackendEngineCfgDescriptor::mEngineDescriptor` will be invalid after copying, so I suggest to `delete` copying methods.

Reply:
@scerzh @amberhassaan 
> the value of `BackendEngineCfgDescriptor::mEngineDescriptor` will be invalid after copying

Maybe I do not understand the whole picture, but shall we also disable copy ctor and operator= in `BackendEngineCfgDescriptor` then?

Reply:
@atamazov No, `BackendEngineCfgDescriptor` is copyable because the value of `BackendEngineCfgDescriptor::mEngineDescriptor` is provided by a user (who created the Engine descriptor).

`OwnedEngineCfgDescriptor` contains the Engine descriptor (as a member) and that pointer should always point to it.

Awkward solution, but not many lines of code.

Reply:
Now I think move c-tor and move `operator=` also don't work
```suggestion
        OwnedEngineCfgDescriptor(const OwnedEngineCfgDescriptor& other)     = delete;
        OwnedEngineCfgDescriptor(OwnedEngineCfgDescriptor&& other) noexcept = delete;
        ;
        OwnedEngineCfgDescriptor& operator=(const OwnedEngineCfgDescriptor& other) = delete;
```
or may be better to undo this chunk and the corresponding one in cpp-file.